src/control/server/mgmt_client.go

```diff
@@ -107,31 +128,27 @@ func (msc *mgmtSvcClient) Join(ctx context.Context, req *mgmtpb.JoinReq) (resp *
 			defer msc.log.Debugf(prefix + " end")
 
 			for {
+				var err error
+				var status *int32
```

Comment:
Why is this a separate pointer variable? Seems like it would be more straightforward to just look at resp.Status, no?

Reply:
but then you have to add extra logic outside of the helper, can do that if you would prefer

Reply:
I'm probably just being dense, but I don't understand why the pointer is used at all. A comment explaining the reasoning would be helpful.